devito/types/dense.py

```diff
@@ -934,6 +919,39 @@ def __init__(self, *args, **kwargs):
             # Dynamically add derivative short-cuts
             self._fd = generate_fd_shortcuts(self)
 
+            # Flag wether it is a parameter or a variable
+            # Used at operator evaluation to evaluate the Function at the
+            # variable location (i.e. if the variable is staggered in x the
+            # parameter has to be computed at x + hx/2)
+            self._is_parameter = kwargs.get('parameter', False)
+
+    @property
+    def is_parameter(self):
+        return self._is_parameter
+
+    def eval_at(self, var):
+        if not self.is_parameter or self.staggered == var.staggered:
+            return self
+
+        return self.subs({d1: var.ind_map[d1] for d1 in self.index_ref})
+
+    @classmethod
+    def staggered_indices(cls, *dimensions, **kwargs):
```

Comment:
I don't think this need to be a separate method. We already have far too many. Since, AFAICT, it's never subclassed, we should/could just squash it with `__indices_setup__`

Reply:
reminder